app/ensure_admin_user.go

```diff
@@ -0,0 +1,143 @@
+package app
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docker/docker/pkg/reexec"
+	"github.com/pkg/errors"
+	"github.com/rancher/rancher/pkg/api/customization/authn"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/urfave/cli"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func RegisterEnsureDefaultAdminCommand() {
+	reexec.Register("/usr/bin/ensure-default-admin", ensureDefaultAdmin)
+	reexec.Register("ensure-default-admin", ensureDefaultAdmin)
+}
+
+func ensureDefaultAdmin() {
+	app := cli.NewApp()
+	app.Description = "Ensure an available default admin user"
+
+	app.Action = func(c *cli.Context) error {
+		kubeConfigPath := os.ExpandEnv("$HOME/.kube/config")
+		if _, err := os.Stat(kubeConfigPath); err != nil {
+			kubeConfigPath = ""
+		}
+
+		conf, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+		if err != nil {
+			return fmt.Errorf("Couldn't get kubeconfig. %v", err)
+		}
+
+		client, err := v3.NewForConfig(*conf)
+		if err != nil {
+			return errors.Errorf("Couldn't get kubernetes client. %v", err)
+		}
+
+		set := labels.Set(map[string]string{"authz.management.cattle.io/bootstrapping": "admin-user"})
+		admins, err := client.Users("").List(v1.ListOptions{LabelSelector: set.String()})
+		if err != nil {
+			return errors.Errorf("Couldn't get default admin user. %v", err)
+		}
+
+		count := len(admins.Items)
+		if count > 1 {
+			var users []string
+			for _, u := range admins.Items {
+				users = append(users, u.Name)
+			}
+			return errors.Errorf("%v users were found with %v label. They are %v. Can only reset the default admin password when there is exactly one user with this label.",
+				count, set, users)
+		} else if count == 1 {
+			admin := admins.Items[0]
+			err = ensureAdminIsEnabled(client, admin)
+			if err != nil {
+				return errors.Errorf("Couldn't enable existing admin. %v", err)
+			}
+			err = ensureAdminIsAdmin(client, admin)
+			if err != nil {
+				return errors.Errorf("Couldn't make existing \"admin\" an actual admin. %v", err)
+			}
+		} else {
+			err = createNewAdmin(client, length)
+			if err != nil {
+				return errors.Errorf("Couldn't create a new admin. %v", err)
+			}
+		}
+
+		return err
+	}
+
+	err := app.Run(os.Args)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func createNewAdmin(client v3.Interface, length int) error {
+	pass := generatePassword(length)
+	hashedPass, err := authn.HashPasswordString(string(pass))
+	if err != nil {
+		return err
+	}
+
+	admin, err := client.Users("").Create(&v3.User{
+		ObjectMeta: v1.ObjectMeta{
+			GenerateName: "user-",
+			Labels:       defaultAdminLabel,
+		},
+		DisplayName:        "Default Admin",
+		Username:           "admin",
+		Password:           string(hashedPass),
+		MustChangePassword: true,
```

Comment:
since the password we are generating is complex and strong and random, no need to require them to change it again. Can set this to false